skimage/util/dask_compatibility.py

```diff
@@ -0,0 +1,493 @@
+    Dask arrays can be passed without conversion back into NumPy objects.
+    """
+    if isinstance(arr_like, da.core.Array):
+        return arr_like
+    else:
+        return np.asanyarray(arr_like)
+
+
+def asarray(arr_like):
+    """
+    Convert array-like input to array, passing dask arrays through.
+
+    Parameters
+    ----------
+    arr_like : array-like
+        Array-like input object to be converted to an array. Can be a NumPy
+        array, a ``dask.array``, or any other array-like object.
+
+    Returns
+    -------
+    arr : array
+        If ``arr_like`` was a ``dask.array`` object it is passed unmodified,
+        otherwise a NumPy array is returned.
+
+    Notes
+    -----
+    This function is intended to replace ``np.asarray`` in the package so
+    Dask arrays can be passed without conversion back into NumPy objects.
+    """
+    if isinstance(arr_like, da.core.Array):
+        return arr_like
+    else:
+        return np.asarray(arr_like)
+
+
+def dtype_conversion_inplace(arr, new_dtype):
+    """
+    Convert dtype of NumPy or dask array inplace.
+
+    Parameters
+    ----------
+    arr : dask or NumPy array
+        Input array, NumPy or Dask arrays supported.
+    new_dtype : NumPy dtype object
+        Valid destination dtype for ``arr``.
+
+    Returns
+    -------
+    arr : dask or NumPy array
+        Output array, converted to dtype ``new_dtype``.
+
+    Notes
+    -----
+    This function is intended to replace ``np.array(..., copy=False)`` as
+    Dask has no equivalent functionality.
+    """
+    # There is no `array` method in dask, so the code must branch
+    if isinstance(arr, np.ndarray):
+        return np.array(arr, new_dtype, copy=False)
+    else:
+        return arr.astype(new_dtype)
+
+
+def eager_min(arr, *args, **kwargs):
+    """
+    Calculate the minimum of a NumPy or Dask array.
+
+    Parameters
+    ----------
+    arr : Dask or NumPy array
+        Input array, NumPy or Dask arrays supported.
+    axis : int, optional
+        Axis along which to operate. By default, flattened input is used.
+    out : ndarray, optional
+        Alternative output array in which to place the result. Must
+        be of the same shape and buffer length as the expected output.
+        See ``doc.ufuncs`` (Section "Output arguments") for more details.
+    keepdims : bool, optional
+        If this is set to True, the axes which are reduced are left
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the original ``arr``.
+
+    Returns
+    -------
+    amax : ndarray or scalar
+        Maximum of `a`. If `axis` is None, the result is a scalar value.
+        If `axis` is given, the result is an array of dimension
+        ``a.ndim - 1``.
+    """
+    amin = arr.min(*args, **kwargs)
+
+    if isinstance(arr, da.core.Array):
+        amin = amin.compute()
```

Comment:
Is `np.array` free on numpy arrays?  If so a common practice here might be to depend on the `__array__` protocol like 

    return np.array(arr.min(*args, **kwargs))

This has the benefit of also supporting other array-like things